akka-actor/src/main/scala/akka/actor/ActorCell.scala

```diff
@@ -727,15 +727,25 @@ private[akka] class ActorCell(
       throw new IllegalArgumentException("ActorCell has no props field")
   }
 
-  final protected def clearActorFields(actorInstance: Actor, recreate: Boolean): Unit = {
-    setActorFields(actorInstance, context = null, self = if (recreate) self else system.deadLetters)
+  final protected def clearActorFields(
+      actorInstance: Actor,
+      recreate: Boolean,
+      context: Option[ActorContext] = Some(null)): Unit = {
+    setActorFields(actorInstance, context = context, self = if (recreate) self else system.deadLetters)
     currentMessage = null
     behaviorStack = emptyBehaviorStack
   }
 
-  final protected def setActorFields(actorInstance: Actor, context: ActorContext, self: ActorRef): Unit =
+  final protected def clearActorFieldsOnTerminate(actorInstance: Actor): Unit =
+    clearActorFields(actorInstance, recreate = false, context = None)
+
+  final protected def setActorFields(actorInstance: Actor, context: Option[ActorContext], self: ActorRef): Unit =
```

Comment:
Use `OptionVal` instead to avoid one alloc and reference-follow.

Reply:
`OptionVal` treats `Some(null)` as `None`, so I've specialised the code path to clear the fields on termination instead.